libs/types/src/models/submission/validation-result.type.ts

```diff
@@ -1,37 +1,43 @@
 export type ValidationResult<T extends object = object> = {
   /**
    * Non-blocking issues with the validated submission.
-   *
-   * @type {ValidationMessage[]}
    */
   warnings?: ValidationMessage<T>[];
 
   /**
    * Blocking issues with the validated submission.
-   *
-   * @type {ValidationMessage[]}
    */
   errors?: ValidationMessage<T>[];
 };
 
-export type ValidationMessage<T extends object> = {
+export type ValidationMessage<
+  T extends object,
+  ID extends keyof ValidationMessages = keyof ValidationMessages
+> = {
   /**
-   * Intl message id.
-   * @type {string}
+   * Localization message id.
    */
-  id: string;
+  id: ID;
 
   /**
    * Associates the message to a input field.
-   *
-   * @type {(keyof T | undefined)}
    */
   field?: keyof T;
 
   /**
-   * Potential values to fill in the message.
-   *
-   * @type {(Record<string, string | number>)}
+   * Values to fill in the message.
    */
-  values?: Record<string, string | number>;
+  values: ValidationMessages[ID];
+};
+
+/**
+ * Map containing validation id as key and values as value
+ */
+export type ValidationMessages = {
```

Comment:
No, the should be defined in ui side for proper translation. Because if we use tran in backend then after switching locales errors and warnjngs will remain unchanged.

Maybe i should add a way to define custom error/warning messages per site to keep them independent, but as i saw in v3 most of them repeats too much often. Maybe add a way to either define them per site and globally

Reply:
Example:
```tsx

import { Trans } from '@lingui/macro';
import { TranslationMessages } from '../../components/translations/translation';

declare module '@postybirb/types' {
  interface ValidationMessages {
    'discord.webhook.name': {
      name: string;
    };
  }
}

TranslationMessages['discord.webhook.name'] = (props) => {
  const { name } = props.values;
  return <Trans>Webhook name {name} is invalid</Trans>;
};

export {};
```